dist/tools/git/git-cache

```diff
@@ -17,22 +21,50 @@ init() {
 }
 
 add() {
+    local repo="$1"
+
+    _locked "$GIT_CACHE_DIR/$name.addlock" _add "$repo"
+}
+
+_add() {
     set -e
     local repo="$1"
     local name="$(_remote_name $repo)"
 
     if ! is_cached "$repo"; then
         git_cache remote add "$name" "$repo"
+        git_cache config --add remote.${name}.fetch "+refs/tags/*:refs/tags/${name}/*"
     else
         echo "git-cache: $url already in cache"
     fi
     set +e
 }
 
+_locked() {
```

Comment:
imagine

if not_in_cache
add
update

called simultaneously might add+update twice, and the second would fail, or just cause an unnecessary update.